src/configuration/app/source-operations.md

```diff
@@ -0,0 +1,64 @@
+---
+search:
+    keywords: ['.platform.app.yaml', 'automated', 'code', 'source', 'update', 'operation', 'dependencies', 'auto']
+---
+
+# Source operations
+
+An application can define a number of operations that apply to its source code and that can be automated.
+
+A basic, common source operation could be to automatically update Composer dependencies like this:
+
+```yaml
+source:
+    operations:
+        update:
+            command: |
+                composer update
+                git add composer.lock
+                git commit -m "Update Composer dependencies."
+```
+
+The `update` key is the name of the operation. It is arbitrary, and multiple source operations can be defined.
+
+The environment resource gets a new `source-operation` action which can be triggered by the CLI:
+
+```
+platform source-operation:run update
+```
+
+The command has two parameters:
+
+* `operation` (string): the name of the operation
+* `variables` (optional, object): additional variables (in the style of variables defined in the `.platform.app.yaml`) to inject in the environment of the source operation
+
+When this operation is triggered:
+
+* A clean Git checkout of the current environment HEAD commit is created; this checkout doesn't have any remotes, has all the tags defined in the project, but only has the current environment branch.
+* Sequentially (_the sequence is in lexicographic order of the path of the `.platform.app.yaml` file_), for each application that has defined this operation, the operation command is launched in the container image of the application, with the same type of environment you will find in a build, with the addition of all the runtime environment variables of the current environment, optionally overridden by the variables specified in the operation payload.
```

Comment:
maybe mention not `$PLATFORM_RELATIONSHIPS`?

Reply:
Reworded that bullet to not be as clumsy, and listed only 2 extra env vars that are available.